### ISSUE TITLE
Arks Layer plugins not working

### ISSUE THREAD

**Issue by JaneSmith** [Yen/PSRT.Astra#27]
With the NGS / 64-bit engine update a while back, the Arks Layer plugins became broken. This was the case for all users, regardless of which launcher they were using. However, the plugins have recently been fixed for PSO2 Tweaker users, [as of a few days ago](https://old.reddit.com/r/PSO2/comments/nnon35/tweaker_update_6044_5282021_jp/). Sadly, they are still broken with Astra for whatever reason. When launching the game, Astra reports that "An error occurred while verifying the Arks-Layer tools". It would be great if this could be fixed soon, or at least before the NGS release on the 9th. Thanks.

**Comment by Yen** [comment]
Hey, thanks for taking an interest Astra.

I am aware of the release of the new arkslayer plugins and am happy they have a new method. I decided a few days ago that I would not move to update astra for the future versions of the game and maintain it going forwards (you can read this now in the readme).

This issue can remain open for if someone else wishes to maintain this project in the future, but support from it by me has ceased.

Sorry for this inconvenience.

**Comment by JaneSmith** [comment]
That's immensely disappointing... I've used Astra since it was released and it's massively better than PSO2 Tweaker. Thanks for letting me know though.

I had the impression that fixing this would be a simple task, as my impression was Astra just downloads the plugin files and puts them in the PSO2 directory, so it would just be a matter of downloading the latest working plugin files. I suppose it's actually more complicated than that though?

**Comment by Yen** [comment]
It might not be a complicated fix, but its likely some larger changes will come with NGS for the JP client just from my experience with the NA beta on the microsoft store. Astra can likely be adapted without too much effort but its a constant series of changes between the game itself and the plugin providers. The issue for me is more the fact that I would need to continue support the next time something changed and so on.

As for this issue itself, the NGS plugins as they are referred to in the tweaker use a new URL but the same format. The switch of endpoint and changes to what plugin files are downloaded should be most of it.

The more important factor however is that PSO2 no longer uses ddraw/pso2h to load plugins, they instead ship a d3d9 shim (pso2/gg delete ddraw.dll if its in the game directory), so the code expecting that behaviour would need to be changed also.

**Comment by ChrisTucker2000** [comment]
I came here looking for news about an NGS update since that's coming out soon, and just saw this. Wow, what a shame. That's really sad to hear this has been abandoned. I hope someone else can pick this up.

**Comment by JaneSmith** [comment]
Yeah, it is sad isn't it. Although from looking at the Readme changes it seems this has been dropped because of PSO2 global. That makes sense. Yen can't expected to keep updating this for a game he/she doesn't even play anymore. I'm glad I was able to use it up to this point at least.

I'm not sure what the note about "changes/improvements to other tools" is supposed to mean though. PSO2 Tweaker is just as bad as it's always been...

I wonder if it would be possible to just use PSO2 Tweaker to download the plugins (and occasionally update them), and then Astra to update and launch the game? Would that still work and let the plugins function?

Edit: I also wonder if maybe the Arks Layer people would be willing to adopt this project (since it's obviously better), and fix the plugins themselves... I suppose they probably wouldn't. Just a thought...
